src/mantine-core/src/components/Select/Select.tsx

```diff
@@ -254,6 +254,12 @@ export const Select = forwardRef<HTMLInputElement, SelectProps>(
       }
     }, [_value]);
 
+    useEffect(() => {
+      if (selectedValue && (!searchable || !dropdownOpened)) {
+        handleSearchChange(selectedValue.label);
+      }
+    }, [selectedValue?.label]);
```

Comment:
I targeted the effect trigger to directly the currently selected item's label. This way we avoid any unnecessary re-renders (such as when the data array reference has changed but the actual data is the same, or when another label not currently selected changes)